eng/packages/http-client-csharp/generator/Azure.Generator/src/ResourceBuilder.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Generator.Mgmt.Models;
+using Azure.Generator.Utilities;
+using Microsoft.TypeSpec.Generator.Input;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.Generator
+{
+    internal class ResourceBuilder
+    {
+        private Dictionary<string, (RequestPath, InputClient)> _specNameToResourceClientMap;
+
+        public bool IsResource(string name) => _specNameToResourceClientMap.ContainsKey(name);
+
+        private IEnumerable<RequestPath>? _resourcePaths;
+        public IEnumerable<RequestPath> ResourcePaths => _resourcePaths ??= _specNameToResourceClientMap.Values.Select(x => x.Item1);
+
+        public ResourceBuilder()
+        {
+            _specNameToResourceClientMap = EnsureOperationsetMap();
+        }
+
+        public IReadOnlyList<(InputClient ResourceClient, string RequestPath, bool IsSingleton, string RequestType, string SpecName)> BuildResourceClients()
+        {
+            var result = new List<(InputClient ResourceClient, string RequestPath, bool IsSingleton, string RequestType, string SpecName)>();
+            var singletonDetection = new SingletonDetection();
+            foreach ((string schemaName, (RequestPath requestPath, InputClient client)) in _specNameToResourceClientMap)
+            {
+                var resourceType = ResourceDetection.GetResourceTypeFromPath(requestPath);
+                var isSingleton = singletonDetection.IsSingletonResource(client, requestPath);
+                result.Add((client, requestPath, isSingleton, resourceType, schemaName));
+            }
+            return result;
+        }
+
+        private Dictionary<string, (RequestPath ResourcePath, InputClient Client)> EnsureOperationsetMap()
+        {
+            var pathToOperationSetMap = CategorizeClients();
```

Comment:
ok - looks like now we never generate operations by operation groups even in "rest operations" class, I like this a lot.

Reply:
well - no - we still may have the issue that our rest operations are still organized by input clients instead of resources.
But in the future if we could get the extra information about resources from typespec side, this should be resolved automatically therefore this is fine now.

Reply:
Let's say we will have resource across inputClients, we can change the output to` Dictionary<string, (RequestPath ResourcePath, IReadOnlyList<InputClient> Clients)`.
The point is to leverage the existing clients without constructing extra objects.